src/value/number.rs

```diff
@@ -107,10 +107,17 @@ impl<'a> fmt::Display for Formatted<'a, Number> {
         }
 
         let mut whole = self.value.value.to_integer().abs();
-        let mut dec = String::new();
+        let mut dec = String::with_capacity(
+            self.value
+                .value
+                .denom()
+                .to_string()
+                .len()
+                .min(self.format.precision),
+        );
```

Comment:
Hm.   Allocating directly to target capacity is good.  But since the point will also be in the string, I think precision + 1 is needed.   The number of chars in a string representation of the rational denominator is probably not relevant, though?  And it cost another string allocation to calculate.  I think just using precision + 1 is better (it's a small enough allocation as well).

```rust
     let mut dec = String::with_capacity(self.format.precision + 1);
```

Reply:
I've updated this [here](https://github.com/kaj/rsass/pull/67/files#diff-1ce7630cffbee60699ecf546bda04efcR112). I agree -- not sure what I was thinking by checking the length of the denominator!